server/JsonStreamingServer/suppliers/JsonStreamingServer.Suppliers.FileStream/Services/CsvHotelOfferReader.cs

```diff
@@ -0,0 +1,81 @@
+﻿using JsonStreamingServer.Core.Models.Domain;
+
+namespace JsonStreamingServer.Suppliers.FileStream.Services
+{
+    public class CsvHotelOfferReader : IDisposable
+    {
+        private readonly string _csvFilePath;
+        private string _csvHeader;
+        private readonly StreamReader _streamReader;
+
+        public CsvHotelOfferReader(string csvFilePath)
+        {
+            _csvFilePath = csvFilePath;
+            _csvHeader = string.Empty;
+            _streamReader = new StreamReader(_csvFilePath);
+        }
+
+        public async Task<HotelOffer?> GetNextOfferAsync(CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrEmpty(_csvHeader))
+            {
+                _csvHeader = await _streamReader.ReadLineAsync(cancellationToken);
+            }
+
+            var line = await _streamReader.ReadLineAsync(cancellationToken);
+
+            if (line == null)
+            {
+                return null;
+            }
+
+            return MapToHotelOffer(line);
+        }
+
+        private HotelOffer MapToHotelOffer(string line)
+        {
+            var fields = line.Split(',');
```

Comment:
Please add a check for the number of fields inside the line and skip it if there are not enough fields.

Reply:
I still hold my comment

Reply:
Added a check on minimun fields amount, safely returning null if not correct